Rebased ww49 hwc  into celadon_master

Shao-Feng opened the pull request:
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

yiyao12 committed:
Fix a logical error when there is only one layer with Client type composing

Change-Id: I2ee0caaf2e9adf96ff3c833a43ee4144137db8ee
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85654
Signed-off-by: Yi Yao <yi.yao@intel.com>

Shao-Feng committed:
Ignore queue update without DRM master.

Ignore displayqueue.QueueUpdate without DRM master.
Since surface can not be committed to DRM without Drm master.
Ignore any update, to avoid wasting GPU resource for
rendering

Change-Id: I7034297b0d30dd7b9c07f0a6d7abd94b4865c225
Tests: No EGL exit in 10 round reboot.
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85425
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Update the alpha blending logic for composing.

When layer alpha become 0xff(blending is Premult), re-validate for
composing or scanout. when the alpha(Premult) become 0, still need
to compose, otherwise, a alpha 255 surface will be shown(for alpha=0)
in scanout.

Change-Id: I97bd1321c5122fc05d7b68cbd23c4fc1692ea7d4
Tests: Work well on Android Q
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85651
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

shuangwan01 committed:
Fixed the dmabuf leak issue caused by prime FDs are not released.

Release prime FDs if the buffer needs to be released.

Change-Id: Id9381a2fbd7681aca71a5e6ab387713f5b644e31
Tracked-On: https://jira.devtools.intel.com/browse/OAM-82969
Tests: HWC virtual memory doesn't inrease abnormally.
Signed-off-by: Wan Shuang <shuang.wan@intel.com>

shuangwan01 committed:
Fixed the HWC crash due to messed virtual display handle.

The virtual display handle tracked in map might be messed due to
newly created display instance might overwrite exiting one.

Tracked-On: None
Tests: HWC doesn't crash when create/destroy VDS frequently.
Signed-off-by: Wan Shuang <shuang.wan@intel.com>

yuanzhel committed:
Fix klocwork issue

Delete an array allocated by new before the function return.

Change-Id: I971c5576a438bac42f9400897ac0fc7ca9df92ce
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85577
Signed-off-by: Yuanzhe Liu <yuanzhe.liu@intel.com>

yuanzhel committed:
Fix screen flicker in Mosaic mode

Add seperate logic for Mosaic mode OverlayLayer Validate.

Change-Id: I4cc3acea6f35d67601a3076dd17de6eb2fd3ff98
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85023
Signed-off-by: Yuanzhe Liu <yuanzhe.liu@intel.com>

Shao-Feng committed:
[Revert me]Assign the commit fence to the first video

As a WA, assign the commit fence to the first video layer.
for CTS testing. and this issue should be fixed in AOSP

Change-Id: I753e4513781d68463fed474478919eb12ae83d1a
Tests: PASS in CTS AdaptivePlayback testing
Tracked-On: https://jira.devtools.intel.com/browse/OAM-85616
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Fix a null pointer in 2 Overlay case

When quiting video playback, the overlay.begin + composition
could be larger than overlay.end.

Change-Id: Ie7fcdee210e75f010a710025c061953307e796c5
Tests: Work well on Acrn with 3 displays.
Tracked-On: https://jira.devtools.intel.com/browse/OAM-86947
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

yuanzhel committed:
Fix klocwork critical issue

Add a null pointer checking before dereference.

Change-Id: I178c0fd9c148e353640f3dfeda7a3b2038e88ad2
Test: Boot successful
Tracked-On: No
Signed-off-by: Yuanzhe Liu <yuanzhe.liu@intel.com>

Shao-Feng committed:
Use merged transform for rotation

The merged transform for layer and plane rotation should be used for
calculating rotated displayframe

Change-Id: Id30e07d06acf158293a1fdadd59a17792f81fa48
Tests: Work well on Android P and Q
Tracked-On: None
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Force to create GL surface for ForceXXXForAllLayers

If the first layer is video, a nativesurface will be created during
ForceGpuForAllLayers. In ForceGpuForAllLayers, all surface should be
GL surface. For VPP composing, the surface should also be 3D format

Change-Id: I0cd48ee863329e64afab4c6497fd2f3d52485f50
Tests: Work well on CIV for video case
Tracked-On: https://jira.devtools.intel.com/browse/OAM-88556
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

dyang23 committed:
replace spin lock with mutex

The spin lock is a dead-loop waiting lock, incase
a realtime thread waiting for it, the cpu resource
won't be released and trigger a dead-lock, so use
mutext to replace it.

update std to C++14 from C++11

Change-Id: I572effc8fa79df1a1df8bc44c9e70bf86e23acd3
Tracked-On: None
Signed-off-by: zhang jun <jun.zhang@intel.com>
Signed-off-by: Yang, Dong <dong.yang@intel.com>

Shao-Feng committed:
Should validate from 0 when current layer is new video

Should validate all layers again when video layer presents and no
correcponding previous layer in previous list.

Change-Id: Ib77b24dd751c0362b27d9159e483d96e924b47c1
Tests: Protected Video can be played well
Tracked-On: https://jira.devtools.intel.com/browse/OAM-88442
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Set commit fence for all video layer

CTS test case not only check the fence of first frame
CTS also check the fence after the video config is changed
has to assign the commit fence for all video layer for
getting pass in CTS.

Change-Id: I7f3a041cad3e77888b871d03ea9fac79919b87f6
Tests: Get PASS in CTS AdaptivePlaybackTest.
Tracked-On: https://jira.devtools.intel.com/browse/OAM-88423
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Skip the client-composed cursor layer

When all layer are set as Client (1 plane case)
the (client)cursor layer still be sent from SF incorrectly.
Skip this duplicated cursor.

Change-Id: Ifa562863f0c7423c014ed6c276dfd416a4989da9
Tests: UI normal, Cursor layer is present well
Tracked-On: None
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

Shao-Feng committed:
Set all layer as Client on CIV

The buffer of Antutu6.1.1-3D can not be scaned-out directly.
WA: set all layer as Client on KVMgt

Change-Id: Ib7e4c20a1e05cf9dcc8e690373c20eac0c7ab042
Tests: UI normal
Tracked-On: https://jira.devtools.intel.com/browse/OAM-88830
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

sysopenci reviewed (commented):
Autobuild started from pull-request-changes on this PR.

FAILURE: CheckBug Bad comments/Bugs

For more information, see: /absp/builders/celadon-autobuild/builds/1447

Shao-Feng commented:
please help to review, thanks

Shao-Feng committed:
Enable Double-Buffering for CIV

If double-buffering is not enable, hwc spend a lot of time
in waiting for previous fence before commiting.

Change-Id: I2cbe7337e04bc0803fa9ddc4d3d7b7ec3c402990
Tests: No obvious frame is missing in video playback on CIV
Tracked-On: https://jira.devtools.intel.com/browse/OAM-88768
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>

sysopenci reviewed (commented):
Autobuild started from pull-request-changes on this PR.

FAILURE: CheckBug Bad comments/Bugs

For more information, see: /absp/builders/celadon-autobuild/builds/1469

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckApprovals Some checks are failed. See the step results.html

More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/1358

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/1360

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckBug Bad comments/Bugs

More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/1361

sysopenci reviewed (commented):
PR integrated into 'preintegration' builder



More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/1362

Done by "Lu, Yang A" <yang.a.lu@intel.com>

sysopenci reviewed (commented):
PR integrated into 'premerge' builder



More details, logs, and binaries: /absp/builders/celadon-premerge/builds/672

Done by "Lu, Yang A" <yang.a.lu@intel.com>

sysopenci reviewed (commented):
cactus finished building and testing your changes 

It was requested for merge by "Lu, Yang A" <yang.a.lu@intel.com> 

so we merge it. Thanks for your contribution.

more details, logs, and binaries: /absp/builders/celadon-mergerequest/builds/1367
